src/odb/src/db/dbBlock.cpp

```diff
@@ -2211,6 +2239,20 @@ dbExtControl* dbBlock::getExtControl()
   return (block->_extControl);
 }
 
+dbDft* dbBlock::createDft()
+{
+  _dbBlock* block = (_dbBlock*) this;
+  _dbDft* dft = block->_dft_tbl->create();
+  block->_dft = dft->getId();
+  return (dbDft*) dft;
+}
```

Comment:
Can you have more than one dft per block?  It doesn't seem that way but this code would allow repeated calls.

Reply:
Just to be clear, the issue that you are seeing here is that we may create more than one dft per block (agree that's an issue), not that I am using a table (_dft_tbl) to hold the pointer right?

Reply:
If there can only be one then there is no need for a table.  The table optimizes allocation of many objects.  Its not too harmful (we allocate a page of objects but only use one) but I'm sure it is helpful

Reply:
So, the tricky thing is that I found no way of setting the owner to my dbDft class without using the dbTable as a "wrapper". I looked into other objects inside the block and I don't think they use a pattern similar to that. However, I changed the code and now we will only have one DFT object per block and the user of odb will not be able to create more.